pkg/services/govmomi/vcenter/clone.go

```diff
@@ -151,8 +151,16 @@ func Clone(ctx *context.VMContext, bootstrapData []byte, format bootstrapv1.Form
 
 	deviceSpecs = append(deviceSpecs, networkSpecs...)
 
-	if err != nil {
-		return errors.Wrapf(err, "error getting network specs for %q", ctx)
+	if len(ctx.VSphereVM.Spec.VirtualMachineCloneSpec.PciDevices) != 0 {
```

Comment:
This adds a second codepath which adds PciDevices to a VM. The already existing code is at https://github.com/chrischdi/cluster-api-provider-vsphere/blob/234e7751f7bb09bb438eed6b4feabbc274cae842/pkg/services/govmomi/service.go#L478 .

I'd prefer only having one place to do this so either this here should get removed or the other place. And we have to take care to not break existing use cases.